spec/models/agents/webhook_agent_spec.rb

```diff
@@ -59,6 +59,27 @@
       expect(out).to eq(['Event Created', 201])
     end
 
+    it 'should respond with customized response code if configured with `code` option' do
+      agent.options['code'] = '200'
+      out = agent.receive_web_request({ 'secret' => 'foobar', 'some_key' => payload }, "post", "text/html")
+      expect(out).to eq(['Event Created', 200])
+
+      # Empty string is a valid code
+      agent.options['code'] = ''
+      out = agent.receive_web_request({ 'secret' => 'foobar', 'some_key' => payload }, "post", "text/html")
+      expect(out).to eq(['Event Created', 201])
+    end
+
+    it 'should respond with `201` if the code option is nil or missing' do
+      agent.options['code'] = nil
+      out = agent.receive_web_request({ 'secret' => 'foobar', 'some_key' => payload }, "post", "text/html")
+      expect(out).to eq(['Event Created', 201])
+
+      agent.options.delete('code')
```

Comment:
Failures are easier to track done when every test for `nil`, `''` or the absense of the option has its own `it` block but its not required.